app/assets/javascripts/polaris_view_components/tooltip_controller.js

```diff
@@ -0,0 +1,73 @@
+import { Controller } from "@hotwired/stimulus";
+import { createPopper } from "@popperjs/core/dist/esm";
+
+export default class extends Controller {
+  static targets = ["template"];
+  static values = { active: Boolean, position: String }
+
+  initialize() {
+    this.shownTooltip = null;
+  }
+
+  getOffset() {
+    switch (this.positionValue) {
+      case "top", "bottom":
+        return [0, 8]
+      case "left", "right":
+        return [0, 6]
+      default:
+        return [0, 8]
+    }
+  }
+
+  show(event) {
+    if (!this.activeValue) return;
+
+    const popperOptions = {
+      placement: this.positionValue,
+      modifiers: [
+        {
+          name: "matchReferenceSize",
+          enabled: true,
+          fn: ({ state, instance }) => {
+            const widthOrHeight =
+              state.placement.startsWith("left") ||
+                state.placement.startsWith("right")
+                ? "height"
+                : "width";
+            const popperSize = state.rects.popper[widthOrHeight];
+            const referenceSize = state.rects.reference[widthOrHeight];
+            if (popperSize >= referenceSize) return;
+
+            state.styles.popper[widthOrHeight] = `${referenceSize}px`;
+            instance.update();
+          },
+          phase: "beforeWrite",
+          requires: ["computeStyles"]
+        },
+        {
+          name: 'offset',
+          options: {
+            offset: this.getOffset(),
+          },
+        }
+      ],
+    };
+
+    const element = event.currentTarget;
+
+    let tooltip = document.createElement("div");
+    tooltip.className = "Polaris-Tooltip"
+    tooltip.innerHTML = this.templateTarget.innerHTML;
+
+    this.shownTooltip = element.appendChild(tooltip);
+
+    this.popper = createPopper(element, this.shownTooltip, popperOptions);
```

Comment:
It will create popover on every hover. Maybe it make sense to create it once during initialization and then show/hide only like we do with Popover? https://github.com/baoagency/polaris_view_components/blob/main/app/assets/javascripts/polaris_view_components/popover_controller.js

Reply:
Good idea, let me refactor it a bit

Reply:
The original reason for doing it like this, is if you have an index page with 500 orders on it (which I do), and each of those orders needs 3 possible tooltips, you'll be initializing 1500 tooltips, most for no reason. So creating them when needed has proven to be a more performant way in the past, as it's irrelevant how many "might" be needed.

Reply:
Thank you @k0va1.. I'll leave final call to @kirillplatonov 👍

Reply:
I think @forsbergplustwo provided great use case which I didn't considered. Let's keep it as is then 🙂